### PR TITLE
#34 Validator/Check XSS for Title & Static content

### PR DESCRIPTION
**Pull request by trinhlaosu**
#34 Validator/Check XSS for Title & Static content:
- Add package react-hook-form for check validator
- Add package xss for check xss
- Check validator for Title: Length from 1 to 256 characters
- Check xss for Title/Static Content

### CODE REVIEW

File: dmd-frontend/src/components/Editor.tsx [hunk 1]
```diff
@@ -13,51 +16,112 @@ const useStyles = makeStyles((theme: Theme) => ({
...
+    mode: "onChange"
+  });
+  const createDOMPurify = require('dompurify');
+  const purifyDOM = createDOMPurify(window);
 
-  const onFormSubmit = (e: React.ChangeEvent<HTMLFormElement>) => {
+  const onFormSubmit = React.useCallback((data, e) => {
     e.preventDefault();
-    onSaveFunc(text);
+    onSaveFunc(purifyDOM.sanitize(titleValue), purifyDOM.sanitize(contentValue));
```
Comment (chiro-hiro) [inline dmd-frontend/src/components/Editor.tsx:53]: If a person want to add HTML tag in his markdown, does this remove these code?

E.g:
```
How to use <script> tag
```

Comment (trinhlaosu) [reply dmd-frontend/src/components/Editor.tsx:53]: I'm sorry, I'm wrong
DOMpurify only applies to Title
I will fix it now

Comment (chiro-hiro) [reply dmd-frontend/src/components/Editor.tsx:53]: What if the title is the same to example from above?

Comment (trinhlaosu) [reply dmd-frontend/src/components/Editor.tsx:53]: Ok, I got it
I have not thought that Title needs this case
I will research how to encode

**Comment by trinhlaosu** [comment]
> @trinhlaosu Please fill missing field of issue.
Thank you

**Pull request closed without merging**